modules/dhcp/providers/server/isc.rb

```diff
@@ -78,13 +78,13 @@ def parse_config_and_leases_for_records
             opts.merge!(parse_record_options(data))
           end
         end
-        begin
-          subnet = @service.find_subnet(opts[:ip])
-          next unless subnet
-          ret_val << Proxy::DHCP::Reservation.new(opts.merge(:subnet => subnet))
-        rescue Exception => e
-          logger.warn "skipped #{hostname} - #{e}"
+        if opts[:deleted]
+          ret_val.each { |rec| ret_val.delete(rec) if rec.options[:hostname] == hostname }
```

Comment:
```
next if ret_val.delete_if { |rec| rec.options[:hostname] == hostname }
```

Do you think this would be preferable so we raise an exception if we don't match anything to remove? Do you think adding a debug log makes sense here?